db/sltdbt.c

```diff
@@ -169,7 +176,16 @@ static int handle_op_block(struct ireq *iq)
         rc = ERR_NOT_DURABLE;
     }
 
-    if (rc == RC_INTERNAL_RETRY) {
+    if (rc == 0) {
+        /* Calculate average runtime of last 16 toblock calls, and cap it
+           at 25 milliseconds. The value is used to determine how much time we
+           sleep before retrying on a deadlock. It does not have to be 100%
+           accurate so we do not hold a lock here. We could make the number of
+           calls and the cap tunables but for now leave them hardcoded. */
+        avg_toblock_us += ((stopus - startus - avg_toblock_us) >> 4);
```

Comment:
Since avg_toblock_us is global, maybe we should get a local copy here, perform your average and checks, and then update global.

Reply:
It'd be too much a micro-optimization without an actual gain.